src/config_classes/Formatter.gd

```diff
@@ -0,0 +1,149 @@
+	set(new_value):
+		if xml_add_trailing_newline != new_value:
+			xml_add_trailing_newline = new_value
+			emit_changed()
+
+@export var xml_shorthand_tags := ShorthandTags.ALWAYS:
+	set(new_value):
+		if xml_shorthand_tags != new_value:
+			xml_shorthand_tags = new_value
+			emit_changed()
+
+@export var xml_shorthand_tags_space_out_slash := false:
+	set(new_value):
+		if xml_shorthand_tags_space_out_slash != new_value:
+			xml_shorthand_tags_space_out_slash = new_value
+			emit_changed()
+
+@export var xml_pretty_formatting := false:
+	set(new_value):
+		if xml_pretty_formatting != new_value:
+			xml_pretty_formatting = new_value
+			emit_changed()
+
+@export var xml_indentation_use_spaces := false:
+	set(new_value):
+		if xml_indentation_use_spaces != new_value:
+			xml_indentation_use_spaces = new_value
+			emit_changed()
+
+@export var xml_indentation_spaces := 2:
+	set(new_value):
+		if xml_indentation_spaces != new_value:
+			xml_indentation_spaces = new_value
+			emit_changed()
+
+@export var number_remove_leading_zero := true:
+	set(new_value):
+		number_remove_leading_zero = new_value
+		emit_changed()
+
+@export var number_use_exponent_if_shorter := true:
+	set(new_value):
+		number_use_exponent_if_shorter = new_value
+		emit_changed()
+
+@export var color_use_named_colors := NamedColorUse.WHEN_SHORTER:
+	set(new_value):
+		if color_use_named_colors != new_value:
+			color_use_named_colors = new_value
+			emit_changed()
+
+@export var color_primary_syntax := PrimaryColorSyntax.THREE_OR_SIX_DIGIT_HEX:
+	set(new_value):
+		if color_primary_syntax != new_value:
+			color_primary_syntax = new_value
+			emit_changed()
+
+@export var color_capital_hex := false:
+	set(new_value):
+		color_capital_hex = new_value
+		emit_changed()
+
+@export var pathdata_compress_numbers := true:
+	set(new_value):
+		pathdata_compress_numbers = new_value
+		emit_changed()
+
+@export var pathdata_minimize_spacing := true:
+	set(new_value):
+		pathdata_minimize_spacing = new_value
+		emit_changed()
+
+@export var pathdata_remove_spacing_after_flags := true:
+	set(new_value):
+		pathdata_remove_spacing_after_flags = new_value
+		emit_changed()
+
+@export var pathdata_remove_consecutive_commands := true:
+	set(new_value):
+		pathdata_remove_consecutive_commands = new_value
+		emit_changed()
+
+@export var transform_list_compress_numbers := true:
+	set(new_value):
+		transform_list_compress_numbers = new_value
+		emit_changed()
+
+@export var transform_list_minimize_spacing := true:
+	set(new_value):
+		transform_list_minimize_spacing = new_value
+		emit_changed()
+
+@export var transform_list_remove_unnecessary_params := true:
+	set(new_value):
+		transform_list_remove_unnecessary_params = new_value
+		emit_changed()
+
+
+func _init(new_title := "") -> void:
+	title = new_title
+	changed.connect(_on_changed)
+
+func modify_title(new_title: String) -> void:
+	title = new_title
+	emit_changed()
+
+func _on_changed() -> void:
+	GlobalSettings.save()
```

Comment:
You're saving the GlobalSettings while they are being parsed here, resulting in all the weird errors.

Reply:
This also means that the earlier ConfigFile implementation was perfectly fine by itself, and you _could_ make a pr / issue to get that system back, as it is a tiny bit easier to be edited by another human than Resource files, in my opinion